content/docs/faq-state.md

```diff
@@ -1,106 +1,106 @@
+### Ի՞նչ է `setState`-ը անում {#what-does-setstate-do}
 
-`setState()` schedules an update to a component's `state` object. When state changes, the component responds by re-rendering.
+`setState()`-ը պլանավորում է կոմպոնենտի `state` օբյեկտի թարմացում։ state-ի թարմացումից հետո կոմպոնենտը արտապատկերվում է։
 
-### What is the difference between `state` and `props`? {#what-is-the-difference-between-state-and-props}
+### Ի՞նչ տարբերություն կա `state`-ի և `props`-ի միջև {#what-is-the-difference-between-state-and-props}
 
-[`props`](/docs/components-and-props.html) (short for "properties") and [`state`](/docs/state-and-lifecycle.html) are both plain JavaScript objects. While both hold information that influences the output of render, they are different in one important way: `props` get passed *to* the component (similar to function parameters) whereas `state` is managed *within* the component (similar to variables declared within a function).
+[`props`](/docs/components-and-props.html)-ը («properties»-ի կրճատ տարբերակը) և [`state`](/docs/state-and-lifecycle.html)-ը պարզ օբյեկտներ են։ Չնայած երկուսն էլ պահում են ինֆորմացիա\` որը ազդում է արտապատկերման վրա, դրանք ունեն մեկ կարևոր տարբերություն. `props`-ը *փոխանցվում է* կոմպոնենտին (ինչպես ֆունկցիայի պարամետրերը),  մինչդեռ `state`-ը *տեղակայված է* կոմպոնենտի ներսում (ինչպես փոփոխականները հայտարարվում են ֆունկցիայի ներսում)։
 
-Here are some good resources for further reading on when to use `props` vs `state`:
-* [Props vs State](https://github.com/uberVU/react-guide/blob/master/props-vs-state.md)
-* [ReactJS: Props vs. State](https://lucybain.com/blog/2016/react-state-vs-pros/)
+Ահա մի քանի օգտակար ռեսուրսներ լրացուցիչ ընթերցման համար, թե երբ օգտագործել `props` և երբ `state`։
+* [Props, թե State](https://github.com/uberVU/react-guide/blob/master/props-vs-state.md)<sub>eng</sub>
+* [ReactJS: Props, թե State](https://lucybain.com/blog/2016/react-state-vs-pros/)<sub>eng</sub>
 
-### Why is `setState` giving me the wrong value? {#why-is-setstate-giving-me-the-wrong-value}
+### Ի՞նչու է `setState`-ը վերադարձնում սխալ արժեք {#why-is-setstate-giving-me-the-wrong-value}
 
-In React, both `this.props` and `this.state` represent the *rendered* values, i.e. what's currently on the screen.
+React-ում `this.props`-ը և `this.state`-ը ներկայացնում են *արտապատկերված* արժեքներ, այսինքն\` այն ինչ տվյալ պահին երևում է էկրանին։
 
-Calls to `setState` are asynchronous - don't rely on `this.state` to reflect the new value immediately after calling `setState`. Pass an updater function instead of an object if you need to compute values based on the current state (see below for details).
+`setState`-ի կանչերը անսինխրոն են - `setState` կանչելուց հետո, երբ ուզում եք անմիջապես արտացոլել նոր արժեքը\` հույս մի դրեք `this.state`-ի վրա։ Եթե ուզում եք հաշվարկել արժեքները հիմնվելով ընթացիք state-ի վրա, օբյեկտի փոխարեն `setState`-ին փոխանցեք թարմացնող ֆունկցիա։ (ստորև ներկայացված է մանրամասն օրինակ)։
```

Comment:
ճիշտ տարբերակը

`setState`-ի կանչերը ասինխրոն են. `setState`-ի կանչից անմիջապես հետո մի հիմնվեք `this.state`-ի արտացոլած արժեքի վրա։